src/orion/core/worker/storage_backup.py

```diff
@@ -0,0 +1,473 @@
+# pylint: disable=,protected-access,too-many-locals,too-many-branches,too-many-statements
+"""
+Module responsible for storage export/import
+============================================
+
+Provide functions to export and import database content.
+"""
+import logging
+import os
+
+from orion.core.io.database import DatabaseError
+from orion.core.io.database.pickleddb import PickledDB
+from orion.storage.base import BaseStorageProtocol, setup_storage
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+COL_EXPERIMENTS = "experiments"
+COL_ALGOS = "algo"
+COL_BENCHMARKS = "benchmarks"
+COL_TRIALS = "trials"
+
+COLLECTIONS = {"experiments", "algo", "benchmarks", "trials"}
+EXPERIMENT_RELATED_COLLECTIONS = {"algo", "trials"}
+
+STEP_COLLECT_EXPERIMENTS = 0
+STEP_CHECK_BENCHMARKS = 1
+STEP_CHECK_DST_EXPERIMENTS = 2
+STEP_CHECK_SRC_EXPERIMENTS = 3
+STEP_DELETE_OLD_DATA = 4
+STEP_INSERT_NEW_DATA = 5
+STEP_NAMES = [
+    "Collect source experiments to load",
+    "Check benchmarks",
+    "Check destination experiments",
+    "Check source experiments",
+    "Delete data to replace in destination",
+    "Insert new data in destination",
+]
+
+
+def dump_database(storage, dump_host, name=None, version=None):
+    """Dump a database
+
+    Parameters
+    ----------
+    storage: BaseStorageProtocol
+        storage of database to dump
+    dump_host:
+        file path to dump into (dumped file will be a pickled file)
+    name:
+        (optional) name of experiment to dump (by default, full database is dumped)
+    version:
+        (optional) version of experiment to dump
+    """
+    dump_host = os.path.abspath(dump_host)
+
+    # For pickled databases, make sure src is not dst
+    if hasattr(storage, "_db"):
+        orig_db = storage._db
+        if isinstance(orig_db, PickledDB) and dump_host == os.path.abspath(
+            orig_db.host
+        ):
+            raise DatabaseError("Cannot dump pickleddb to itself.")
+
+    dst_storage = setup_storage({"database": {"host": dump_host, "type": "pickleddb"}})
```

Comment:
What do we do if the destination storage already exists? I would be in favor of raising an error if it does, and adding an option to force overwrite. If forced overwrite, then the dump should be deleted and recreated from scratch to avoid merging storages using `dump`.

Reply:
Ok !